src/App.js

```diff
@@ -0,0 +1,227 @@
+import React, {Component} from 'react';
+import FormContainer from './containers/FormContainer/FormContainer';
+import {BrowserRouter} from 'react-router-dom';
+
+class App extends Component {
+    constructor() {
+        super();
+        this.state = {
+            todos: [
+                // {id:1 content:"Task1", done:false}
+            ]
+        }
+
+
+    }
+
+    // fetching todos from db.json
+    fetchData = (url) => {
+        fetch(url)
+            .then((response) => {
+                console.log(response);
+                return response
+            })
+            .then((response) => response.json())
+            .then((todos) => this.setState({todos: todos}))
+            .catch((error) => console.error(error));
+    };
+
+    //GET TODOS
+
+    fetchDataHandler = () => {
+        this.fetchData('http://localhost:3000/todos')
+    };
+
+
+    componentWillMount = () => {
+        this.fetchData('http://localhost:3000/todos');
+        console.log('component will mount');
+        console.log(this.state.todos);
+    };
+
+
+    // these method belows are used for CRUD of todos on json-server, not state.
+
+    //POST
+
+    postTodoHandler = (content_input) => {
+        fetch('http://localhost:3000/todos', {
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json'
+            },
+            body: JSON.stringify({
+                'content': content_input,
+                'done': false
+            })
+        }).then(res => res.json())
+            .then(console.log())
+            .then(() => this.fetchDataHandler());
+    };
+
+    //PUT
+
+    putTodoContentHandler = (event, todo_id, done_stats) => {
```

Comment:
should pass the todo object, instead of event, todo_id, done_stats. 
```
{
  id: 1,
  content: "adada",
  done: true,
}
```

Reply:
I borrowed this code from udemy courses so I am not sure this is really good code, but is there any other way to get the value of textfield input as the "content" of todo?

In this example: http://todomvc.com/examples/react
you can input the content of todo in each textfield and can update at the same time. 
I set "event" in arguments because I will use it in component and set this method for onChange.

And by  "pass the todo object", you mean like this?

```
let todoObject ={
id: todo.id,
content: currentContent,
done:true
}

putTodoContentHandler(todoObject) =>{

```

Reply:
Yes, you could change the `onChange`.
```
// text field
 <TextField
  className={classes.textField}
  label="Todo"
  value={todo.content}
  onChange={(event) => props.putTodoHandler({ content: event.target.value, id: todo.id, done: todo.done})}
/>
// checkbox
 <Checkbox
  //.. 
  onChange={(event) => props.putTodoHandler({ content: todo.content, id: todo.id, done: !todo.done})}
/>
``` 
then you'll only need one function `putTodoHandler`.
no need a individual function for content and status

Reply:
Now perfectly makes sense! Thank you! I did not know that I can pass an object for arguments, because I have never seen it before in tutorial course.